### PR TITLE
add FAQ

### LINKED ISSUE #480: docs on what fastplotlib is
notes from meeting with @apasarkar

* we should put a few bullet points on the readme that quickly explains what fpl allows you to do
  * rapid prototyping, realtime analysis
  * easy API
  * ship with depedent packages as a standalone distributable - ex: pyinstaller, etc. (this is a medium term goal right now)

* somewhere in the docs we should put a comparison with other libraries, pyqtgraph, vispy, napari, bokeh, matplotlib

### PR DESCRIPTION
**Pull request by clewis7**
closes #480

### CODE REVIEW

File: docs/source/user_guide/faq.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        - interactive visualization via an intuitive and expressive API
+        - rapid prototyping and algorithm design
+        - easy exploration and rendering of large-scale data
+
+.. dropdown:: How does `fastplotlib` relate to `matplotlib`?
+
+    `fastplotlib` is **NOT** related to `matplotlib` in any way.
+
+    These are two completely different libraries with their own APIs and use-cases. The `fastplotlib` library is primarily for *interactive*
+    visualization that runs on the GPU using WGPU.
```
Comment (kushalkolar) [inline docs/source/user_guide/faq.rst:22]: add something along the lines  of "the fastplotlib way of doing things is different from matplotlib. Using fastplotlib is actually more similar to using numpy! See the "How can I learn to use `fastplotlib` section below"

Comment (kushalkolar) [reply docs/source/user_guide/faq.rst:22]: I'm trying to figure out how to stop people from search the docs for "imshow" XD

File: docs/source/user_guide/faq.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    We want `fastplotlib` to be easy to learn and use. To get started with the library we recommend taking a look at our `guide <https://fastplotlib.readthedocs.io/en/latest/user_guide/guide.html>`_ and
+    `examples gallery <https://fastplotlib.readthedocs.io/en/latest/_gallery/index.html>`_.
+
+    In general, if you are familiar with numpy and array notation you will already have a intuitive understanding of interacting
+    with your data in `fastplotlib`. If you have any questions, please do not hesitate to post an issue or discussion forum post.
+
+.. dropdown:: Should I use `fastplotlib` for making publication figures?
+
+    **NO!** `fastplotlib` is not meant for creating *static* publication figures. There are many other libraries that are well-suited
```
Comment (kushalkolar) [inline docs/source/user_guide/faq.rst:34]: remove the bold "No", let's be a bit less aggressive :joy: .

File: docs/source/user_guide/faq.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    for this task.
+
+.. dropdown:: How does `fastplotlib` handle data loading?
+
+    `fastplotlib` is a plotting library and **NOT** a data handling or data loading library. These tasks are outside of the scope of
```
Comment (kushalkolar) [inline docs/source/user_guide/faq.rst:39]: less aggressive XD

**Comment by clewis7** [comment]
added some stuff from roadmap meeting this morning

File: docs/source/user_guide/faq.rst [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    No, `fastplotlib` is not meant for creating *static* publication figures. There are many other libraries that are well-suited
+    for this task.
+
+.. dropdown:: How does `fastplotlib` handle data loading?
+
+    `fastplotlib` is a plotting library and not a data handling or data loading library. These tasks are outside of the scope of
+    the library.
+
+    In general, if your data is an array-like object, `fastplotlib` should be able to use it.
+
```
Comment (kushalkolar) [inline docs/source/user_guide/faq.rst:44]: Add something like "if you have issues using your data objects please don't hesitate to post an issue!

**Review by kushalkolar — CHANGES_REQUESTED** [review]

**Comment by kushalkolar** [comment]
Maybe other FAQ to add:

1. [Do I need a GPU](https://fastplotlib.readthedocs.io/en/latest/user_guide/gpu.html)
2. Does this work in Qt blah blah, link to: https://github.com/fastplotlib/fastplotlib#supported-frameworks and https://fastplotlib.readthedocs.io/en/latest/user_guide/guide.html#using-fastplotlib-interactively

File: docs/source/user_guide/faq.rst [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+What can I do with `fastplotlib`?
+---------------------------------
+
+    `fastplotlib` allows for:
+        - GPU accelerated visualization
+        - interactive visualization via an intuitive and expressive API
+        - rapid prototyping and algorithm design
+        - easy exploration and fast rendering of large-scale data
+        - design, develop, evaluate and ship machine learning models
+
```
Comment (kushalkolar) [inline docs/source/user_guide/faq.rst:20]: add "create visualizations for real-time acquisition systems for scientific instruments, such as cameras and other devices"

File: docs/source/user_guide/faq.rst [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+Do I need a GPU?
+----------------
+
+    Technically no, you can perform limited software rendering on linux using lavapipe.
+    However, `fastplotlib` is intentionally built for realtime rendering using the latest GPU technologies, so we
+    strongly recommend that you use a GPU. With that being said, an integrated GPU should be sufficient for many
+    use-cases.
```
Comment (kushalkolar) [inline docs/source/user_guide/faq.rst:27]: maybe start with "integrated GPUs such as those found in most modern laptops is sufficient for many use cases..."

File: docs/source/user_guide/faq.rst [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+How does `fastplotlib` relate to `matplotlib`?
+----------------------------------------------
+
+    `fastplotlib` is **not** related to `matplotlib` in any way.
+
+    These are two completely different libraries with their own APIs and use-cases. The `fastplotlib` library is primarily for *interactive*
+    visualization that runs on the GPU using WGPU. The `fastplotlib` architecture is completely different from `matplotlib`. Using `fastplotlib`
+    is more akin to using `numpy`. See the "How can I learn to use `fastplotlib`?" section below.
```
Comment (kushalkolar) [inline docs/source/user_guide/faq.rst:36]: expand on this a bit more, ideas from: https://github.com/fastplotlib/fastplotlib/issues/122#issuecomment-1399890462 

The pygfx buffer interface is really unlike anything in matplotlib and other libs which is a major reason for why fpl can have an array-like API for plotting!

File: docs/source/user_guide/faq.rst [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    `examples gallery <https://fastplotlib.readthedocs.io/en/latest/_gallery/index.html>`_.
+
+    In general, if you are familiar with numpy and array notation you will already have a intuitive understanding of interacting
+    with your data in `fastplotlib`. If you have any questions, please do not hesitate to post an issue or discussion forum post.
+
+Should I use `fastplotlib` for making publication figures?
+----------------------------------------------------------
+
+    No, `fastplotlib` is not meant for creating *static* publication figures. There are many other libraries that are well-suited
+    for this task.
```
Comment (kushalkolar) [inline docs/source/user_guide/faq.rst:51]: Add that figures can be exported to PNG

File: docs/source/user_guide/faq.rst [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+How does `fastplotlib` handle data loading?
+-------------------------------------------
+
+    `fastplotlib` is a plotting library and not a data handling or data loading library. These tasks are outside of the scope of
+    the library.
+
+    In general, if your data is an array-like object, `fastplotlib` should be able to use it. However, if you have any problems using your data objects,
```
Comment (kushalkolar) [inline docs/source/user_guide/faq.rst:59]: also add link to related issue: https://github.com/fastplotlib/fastplotlib/issues/483

File: docs/source/user_guide/faq.rst [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+    please do not hesitate to post an issue!
+
+What is the scope of `fastplotlib`?
+-----------------------------------
+
+    While the capabilities are very far-reaching, we would like to emphasize that `fastplotlib` is a general-purpose plotting library focused on scientific visualization.
+    More specifically, we aim to develop the tools necessary for users to build fast and interactive visualizations for a variety of scientific domains (e.g. neuroscience,
+    astrophysics). If you have a particular feature in mind that you feel is missing, please post an issue and we will respond accordingly letting you know if it fits within
```
Comment (kushalkolar) [inline docs/source/user_guide/faq.rst:67]: "including but not limited to neuroscience, astronomy, biology, computer vision, signal processing, etc. "

File: docs/source/user_guide/faq.rst [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+    the scope of the project.
+
+What types of PRs are we willing to accept?
+-------------------------------------------
+
+    Primarily the features of `fastplotlib` have been developed as they relate to the core-developers research use cases (mostly neuroscience). With that being said, there are many domains in which
```
Comment (kushalkolar) [inline docs/source/user_guide/faq.rst:73]: add more to our research use case:

(neuroscience, algorithm development, and machine learning), it's more broad XD

File: docs/source/user_guide/faq.rst [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+    we do not have the knowledge to best-implement the tools needed for proper visualization. We welcome all PRs that address these types of missing functionality. We
+    recommend taking a look at our `Roadmap <https://github.com/fastplotlib/fastplotlib/issues/55>`_ to get a better idea of what those items might be :D
+
+    Closely related to this, we would love to add more examples to our repo for different types of scientific visualizations. We welcome all PRs that showcase using `fastplotlib` for
```
Comment (kushalkolar) [inline docs/source/user_guide/faq.rst:77]: exclamation mark !

File: docs/source/user_guide/faq.rst [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+What frameworks does `fastplotlib` support?
+-------------------------------------------
+
+    The short answer is that `fastplotlib` can run on anything that `pygfx` runs on. This includes,
+        - `jupyter lab` using `jupyter_rfb`
+        - `PyQt` and `PySide`
+        - `glfw`
+        - `wxPython`
+
+    Note: Use in Google Colab does not currently work because Colab does not offer support for `jupyter_rfb`
```
Comment (kushalkolar) [inline docs/source/user_guide/faq.rst:92]: Myself and manz did implement it for colab, but google's cloud systems are slow to the point that it's unusable on colab. Alternative inexpensive cloud provider options which we recommend are CodeOcean and Lambda Cloud, I've tested fpl on these and it works really well!

**Review by kushalkolar — CHANGES_REQUESTED** [review]
this is great :D few comments/adjustments

File: docs/source/user_guide/faq.rst [hunk 14]
```diff
@@ -17,14 +17,14 @@ What can I do with `fastplotlib`?
...
 Do I need a GPU?
 ----------------
 
-    Technically no, you can perform limited software rendering on linux using lavapipe.
-    However, `fastplotlib` is intentionally built for realtime rendering using the latest GPU technologies, so we
-    strongly recommend that you use a GPU. With that being said, an integrated GPU should be sufficient for many
-    use-cases.
+    Integrated GPUs, such as those found in modern laptops is sufficient for many use cases. However, you can perform limited software rendering on linux
+    using lavapipe. With that being said, `fastplotlib` is intentionally built for realtime rendering using the latest GPU technologies, so we
+    strongly recommend that you use a GPU.
```
Comment (kushalkolar) [inline docs/source/user_guide/faq.rst:27]:
```suggestion
    Integrated GPUs, such as those found in modern laptops is sufficient for many use cases. 
    
    For the best performance you will require a dedicated GPU. You can think of it like running a game, a more complex visualization or faster rendering will require a better GPU. 
    
    Limited software rendering using just the CPU is supported on linux using lavapipe, but this is mostly only useful for testing purposes. 
```

File: docs/source/user_guide/faq.rst [hunk 15]
```diff
@@ -48,7 +52,7 @@ Should I use `fastplotlib` for making publication figures?
 ----------------------------------------------------------
 
     No, `fastplotlib` is not meant for creating *static* publication figures. There are many other libraries that are well-suited
-    for this task.
+    for this task. However, `fastplotlib` figures can be exported to PNG using ``figure.export()``.
```
Comment (kushalkolar) [inline docs/source/user_guide/faq.rst:55]: rephrase, start with "While `fastplotlib` figures can be exported to PNG ..., `fastplotlib` is not intended for creating *static* publication figures..."

File: docs/source/user_guide/faq.rst [hunk 16]
```diff
@@ -89,7 +93,8 @@ What frameworks does `fastplotlib` support?
         - `glfw`
         - `wxPython`
 
-    Note: Use in Google Colab does not currently work because Colab does not offer support for `jupyter_rfb`
+    Note: Use in Google Colab is not highly functional. We recommend using an inexpensive alternative cloud provider
+    such as CodeOcean or Lambda Cloud. Kushal has tested these and `fastplotlib` works very well.
```
Comment (kushalkolar) [inline docs/source/user_guide/faq.rst:97]: lmao I'm the scapegoat :joy:

**Review by kushalkolar — CHANGES_REQUESTED** [review]

File: docs/source/user_guide/faq.rst (hunk 16 above)
Comment (clewis7) [reply docs/source/user_guide/faq.rst]: bahaha, I didn't mean it like that...fixed so it is less ambiguous who is to blame if it doesn't work lol

File: docs/source/user_guide/faq.rst [hunk 17]
```diff
@@ -22,9 +22,11 @@ What can I do with `fastplotlib`?
 Do I need a GPU?
 ----------------
 
-    Integrated GPUs, such as those found in modern laptops is sufficient for many use cases. However, you can perform limited software rendering on linux
```
Comment (kushalkolar) [inline docs/source/user_guide/faq.rst]: "Integrated GPUs, such as those found in modern laptops **are** sufficient"

**Review by kushalkolar — APPROVED** [review]

**Pull request merged**